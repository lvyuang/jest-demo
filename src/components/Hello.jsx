import React from 'react'
import styled from 'styled-components'

const Div = styled.div`
  background-color: red;
`

class Hello extends React.Component {
  constructor () {
    super()
    this.state = {
      text: 'Hello World!'
    }
  }

  click = () => {
    this.setState({
      ...this.state,
      text: this.state.text + ' clicked'
    })

    const { onClick } = this.props
    onClick && onClick()
  }

  render () {
    return (
      <Div onClick={this.click}>{this.state.text}</Div>
    )
  }
}

export default Hello