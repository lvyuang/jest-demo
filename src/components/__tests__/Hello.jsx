import React from 'react'
import { shallow, mount } from 'enzyme'

import Hello from '../Hello'

describe('React 组件测试', () => {
  test('shallow render Hello', () => {
    const com = shallow(<Hello />)

    expect(com).toMatchSnapshot()
  })

  test('mount render Hello', () => {
    const com = mount(<Hello />)

    expect(com).toMatchSnapshot()
  })

  test('click Hello', () => {
    const com = mount(<Hello />)

    com.find('div').simulate('click')

    expect(com).toMatchSnapshot()
  })

  test('onClick should be called only once', () => {
    const fakeClick = jest.fn()
    const com = mount(<Hello onClick={fakeClick} />)

    com.find('div').simulate('click')

    expect(fakeClick.mock.calls.length).toBe(1)
  })
})
