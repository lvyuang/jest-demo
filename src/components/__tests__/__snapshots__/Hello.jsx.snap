// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React 组件测试 click Hello 1`] = `
.c0 {
  background-color: red;
}

<Hello>
  <styled.div
    onClick={[Function]}
  >
    <div
      className="c0"
      onClick={[Function]}
    >
      Hello World! clicked
    </div>
  </styled.div>
</Hello>
`;

exports[`React 组件测试 mount render Hello 1`] = `
.c0 {
  background-color: red;
}

<Hello>
  <styled.div
    onClick={[Function]}
  >
    <div
      className="c0"
      onClick={[Function]}
    >
      Hello World!
    </div>
  </styled.div>
</Hello>
`;

exports[`React 组件测试 shallow render Hello 1`] = `
<styled.div
  onClick={[Function]}
>
  Hello World!
</styled.div>
`;
